### ISSUE TITLE
`numpy_helper.from_array` fails with `onnx._custom_element_types` `int4` and `uint4`

### ISSUE THREAD

**Issue by quic-ernst** [onnx/onnx#6227]
# Bug Report

### Is the issue related to model conversion?

No.

### Describe the bug

I saw that ONNX had support with Quant Dequant with Opset 21. In checking this out, I tried playing around with taking a numpy array with a dtype of ONNX's int4 representation (as numpy does not have int4 support) and the `numpy_helper.from_array` fails with this. In looking at the tagged release branch of v1.16.1 and my own python install ONNX version, it looks like this isn't supported and was missed in this release. In the main branch, it looks like there is support for this. I'm wondering if this was missed?

v1.16.1 - helper.py : `np_dtype_to_tensor_dtype`

https://github.com/onnx/onnx/blob/595228d99e3977ac27cb79d5963adda262af99ad/onnx/helper.py#L1588-L1600

main - helper.py : `np_dtype_to_tensor_dtype`

https://github.com/onnx/onnx/blob/c7717bb39c684a6d86c82a5bb1d4c0e5d90353fe/onnx/helper.py#L1594-L1625

---

v1.16.1 - numpy_helper.py : `from_array`

https://github.com/onnx/onnx/blob/595228d99e3977ac27cb79d5963adda262af99ad/onnx/numpy_helper.py#L335-L384

main -  numpy_helper.py : `from_array`

https://github.com/onnx/onnx/blob/c7717bb39c684a6d86c82a5bb1d4c0e5d90353fe/onnx/numpy_helper.py#L475-L533

### System information
- OS Platform and Distribution : Linux Ubuntu 20.04
- ONNX version : 1.16.1
- Python version: 3.10.2
- GCC/Compiler version (if compiling from source): N/A
- CMake version: N/A
- Protobuf version: N/A
- Visual Studio version (if applicable): N/a

### Reproduction instructions
```python
import numpy as np
import onnx
from onnx.reference.custom_element_types import int4, uint4


arr = np.array([1, 0], dtype=int4)
tensor = onnx.numpy_helper.from_array(arr)
```

### Expected behavior

I expect a `TensorProto` of dtype `TensorProto.INT4` to come back after the function call.

**Comment by justinchuby** [comment]
Should be fixed in 1.17. cc @xadupre to confirm

**Comment by quic-ernst** [comment]
@justinchuby Will there be a formal patch release for 1.16 with this or just push to 1.17?
